flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -663,6 +677,9 @@ public boolean receiveAcknowledgeMessage(AcknowledgeCheckpoint message) throws E
 						LOG.info("Completed checkpoint " + checkpointId + " (in " +
 								completed.getDuration() + " ms)");
 
+						lastCheckpointSize = completed.getStateSize();
+						lastCheckpointDuration = completed.getDuration();
```

Comment:
Would it make sense to use the `CheckpointStatsTracker` to obtain the values for the last checkpoint? Given that the JobManager is not so high performance critical with respect to the calculation of the metrics.